packages/utils/plugins-helpers/src/federation.ts

```diff
@@ -28,14 +30,48 @@ export const federationSpec = parse(/* GraphQL */ `
   directive @key(fields: _FieldSet!) on OBJECT | INTERFACE
 `);
 
+interface TypeMeta {
+  hasResolveReference: boolean;
+}
+
+export type FederationMeta = { [typeName: string]: TypeMeta };
+
 /**
- * Adds `__resolveReference` in each ObjectType involved in Federation.
+ * Adds `__resolveReference` in each ObjectType and InterfaceType involved in Federation.
  * @param schema
  */
-export function addFederationReferencesToSchema(schema: GraphQLSchema): GraphQLSchema {
-  return mapSchema(schema, {
+export function addFederationReferencesToSchema(schema: GraphQLSchema): {
```

Comment:
We are transforming the schema to have `__resolveReference` fields to prepare for the resolvers plugin. Because of this pattern, it's worth capturing meta here so, and use the meta instead of running `checkTypeFederationDetails` in the plugin.

Because of this, `checkTypeFederationDetails` is now only used in this file i.e. it's no longer exported and called in resolvers plugin -> better logic encapsulation